xla/service/gpu/transforms/cudnn_fused_conv_rewriter.cc

```diff
@@ -488,83 +564,138 @@ void CaptureConvGraphRecursive(HloInstruction* instr,
+      if (graph_string.AppendOp("add", op0, {operand0, operand1})) {
         operands.push_back(operand0 == instr ? operand1 : operand0);
-        num_linear_users++;
+        ++num_new_users;
         CaptureConvGraphRecursive(user, operands, aux_outputs, graph_string,
-                                  visited_instrs, final_instr);
+                                  visited_instrs, final_instr, num_endpoints);
+      } else {
+        // Since operands only holds ops that are not part of the graph, remove
+        // instr.
+        operands.erase(std::remove(operands.begin(), operands.end(), instr),
+                       operands.end());
+        ++num_existing_users;
       }
       continue;
     }
     // Scale
-    if (Match(user, m::MultiplyAnyOrder(&op, m::Op(&operand0),
+    if (Match(user, m::MultiplyAnyOrder(&op0, m::Op(&operand0),
                                         m::Broadcast(m::Op(&operand1)))) &&
         ShapeUtil::IsScalar(operand1->shape())) {
-      if (graph_string.AppendOp("scale", op, {operand0, operand1})) {
+      if (graph_string.AppendOp("scale", op0, {operand0, operand1})) {
         operands.push_back(operand1);
-        num_linear_users++;
+        ++num_new_users;
         CaptureConvGraphRecursive(user, operands, aux_outputs, graph_string,
-                                  visited_instrs, final_instr);
+                                  visited_instrs, final_instr, num_endpoints);
       }
       continue;
     }
     // Inverse Scale
-    if (Match(user, m::Divide(&op, m::Op(&operand0),
+    if (Match(user, m::Divide(&op0, m::Op(&operand0),
                               m::Broadcast(m::Op(&operand1)))) &&
         ShapeUtil::IsScalar(operand1->shape())) {
-      if (graph_string.AppendOp("invscale", op, {operand0, operand1})) {
+      if (graph_string.AppendOp("invscale", op0, {operand0, operand1})) {
         operands.push_back(operand1);
-        num_linear_users++;
+        ++num_new_users;
         CaptureConvGraphRecursive(user, operands, aux_outputs, graph_string,
-                                  visited_instrs, final_instr);
+                                  visited_instrs, final_instr, num_endpoints);
       }
       continue;
     }
     // ReLU
-    if (Match(user, m::MaximumAnyOrder(&op, m::Op(&operand0),
+    if (Match(user, m::MaximumAnyOrder(&op0, m::Op(&operand0),
                                        m::Broadcast(m::ConstantScalar(0))))) {
-      if (graph_string.AppendOp("relu", op, {operand0})) {
-        num_linear_users++;
+      if (graph_string.AppendOp("relu", op0, {operand0})) {
+        ++num_new_users;
         CaptureConvGraphRecursive(user, operands, aux_outputs, graph_string,
-                                  visited_instrs, final_instr);
+                                  visited_instrs, final_instr, num_endpoints);
       }
       continue;
     }
     //  Maximum of the absolute value (Amax) following ReLU (elided Abs) -- not
     //  a linear user
```

Comment:
The concept of "linear" users don't exist anymore. Instead can mention this is an aux output.

And same for the comment: "Maximum of the absolute value (Amax) -- not a linear user"